src/Protocol/Http/Authenticator/JwtAuthenticator.php

```diff
@@ -30,6 +30,21 @@ class JwtAuthenticator implements AuthenticatorInterface
      */
     private $jwt;
 
+    /**
+     * @var string
+     */
+    private $jws = '';
+
+    /**
+     * @var integer
+     */
+    private $jwsLifetime = 3600;
+
+    /**
+     * @var integer
+     */
+    private $jwsCreatedAt = 0;
```

Comment:
What is created at? You use addition operation for next check if token is valid.

I think, what better - rename this property to `jwsUntilTo` or `jwsValidTo`. After this, we can easy check:

```
if ($this->jwsValidTo <= time()) {
   // Regenerage
}
```